integrations/llama_index/tests/test_toolkit.py

```diff
@@ -176,3 +269,234 @@ def test_toolkit_with_invalid_function_input(client):
+            UCFunctionToolkit(function_names=[func_obj.full_function_name])
+
+
+def test_toolkit_creation_without_properties_argument_mocked():
+    """
+    Test that UCFunctionToolkit successfully creates a tool when the function does not have a 'properties' argument using mocks.
+    """
+    mock_function_info = generate_mock_function_info(has_properties=False)
+
+    mock_client = mock.create_autospec(BaseFunctionClient, instance=True)
+    mock_client.get_function.return_value = mock_function_info
+    mock_client.to_dict.return_value = {"mock": "config"}
+    mock_client.execute_function.return_value = MockFunctionExecutionResult(
+        return_value='{"value": "some_string"}'
+    )
+
+    class MockPydanticModelWithoutProperties(BaseModel):
+        x: str
+
+    mock_fn_schema = mock.Mock()
+    mock_fn_schema.pydantic_model = MockPydanticModelWithoutProperties
+
+    with (
+        mock.patch(
+            "ucai.core.utils.function_processing_utils.generate_function_input_params_schema",
+            return_value=mock_fn_schema,
+        ),
+        mock.patch(
+            "ucai_llamaindex.toolkit.validate_or_set_default_client",
+            return_value=mock_client,
+        ),
+    ):
+        toolkit = UCFunctionToolkit(
+            function_names=["catalog.schema.test_function"], client=mock_client
+        )
+        tools = toolkit.tools
+        assert len(tools) == 1
+        tool = tools[0]
+        assert tool.metadata.name == "catalog__schema__test_function"
+        assert not tool.metadata.return_direct
+        assert tool.metadata.description == "A test function without properties argument"
+        assert tool.client_config == {"mock": "config"}
+
+        input_args = {"x": "some_string"}
+        result = json.loads(tool.fn(**input_args))["value"]
+        assert result == "some_string"
+
+
+def test_uc_function_to_llama_tool_mocked():
+    """
+    Test the conversion of a Unity Catalog function to a Llama tool without 'properties' argument using mocks.
+    """
+    mock_function_info = generate_mock_function_info(has_properties=False)
+
+    class MockPydanticModelWithoutProperties(BaseModel):
+        x: str
+
+    mock_fn_schema = mock.Mock()
+    mock_fn_schema.pydantic_model = MockPydanticModelWithoutProperties
+
+    mock_client = mock.create_autospec(BaseFunctionClient, instance=True)
+    mock_client.get_function.return_value = mock_function_info
+    mock_client.to_dict.return_value = {"mock": "config"}
+    mock_client.execute_function.return_value = MockFunctionExecutionResult(
+        return_value='{"value": "some_string"}'
+    )
+
+    with (
+        mock.patch(
+            "ucai.core.utils.function_processing_utils.generate_function_input_params_schema",
+            return_value=mock_fn_schema,
+        ),
+        mock.patch(
+            "ucai_llamaindex.toolkit.validate_or_set_default_client",
+            return_value=mock_client,
+        ),
```

Comment:
can we validate these mock are used using `assert_called`?

Reply:
yep!

Reply:
great call - we preempt the patch for generating the schema (no need to patch as it is unreachable). Remove the patch (and added call assertions for the client patch)